### DISCUSSION TITLE
Quickstart - how to compile

### DISCUSSION THREAD

**Discussion by Hennich** [jedisct1/libsodium#1419] [General]
The quickstart should have a small section on how to compile the example code.
E.g., that you have to explicitly link the library via `gcc file.c -lsodium`. Compiler commands for other platforms might also be nice.

**Comment by jedisct1** [comment]
There are many ways to add a library to a project, depending on the compiler, IDE, and build system.

If using the Zig build system, the Git repository is directly going to be added as a dependency. 

With `cmake`, `autotools` and other Makefile generators, `-lsodium` is never passed directly either.

With IDEs like Xcode or Visual Studio, this is done by clicking buttons.

Even when being explicit, on many environments, people may also want to use `pkg-config` instead of `-lsodium`.

There's nothing special about libsodium compared to other libraries. It's expected to be added to a project the same as any other library.

So documenting this would be a little bit off-topic, and always inacurate/incomplete/outdated compared to the existing documentation of compilers, build systems and IDEs.
